deploy/local_single_user_deploy.sh

```diff
@@ -42,48 +72,107 @@ generate_config_files() {
-           --argjson new_port_keeper_datastore $new_port_keeper_datastore \
-           --arg i "$i" \
-           '.chrono_keeper.KeeperRecordingService.rpc.service_base_port = $new_port_keeper_record |
+            --arg output_dir "$output_dir" \
+            --argjson new_port_keeper_record $new_port_keeper_record \
+            --argjson new_port_keeper_drain $new_port_keeper_drain \
+            --argjson new_port_keeper_datastore $new_port_keeper_datastore \
+            --argjson recording_group "$recording_group" \
+            --arg j "$j" \
+            '.chrono_keeper.KeeperRecordingService.rpc.service_base_port = $new_port_keeper_record |
+            .chrono_keeper.KeeperGrapherDrainService.rpc.service_base_port = $new_port_keeper_drain |
             .chrono_keeper.KeeperDataStoreAdminService.rpc.service_base_port = $new_port_keeper_datastore |
-            .chrono_keeper.Logging.log.file = ($bin_dir + "/" + $i + "_" + .chrono_keeper.Logging.log.file)' "$default_conf" > "$output_file"
-
-        echo "Generated $output_file with ports $new_port_keeper_record and $new_port_keeper_datastore."
+            .chrono_keeper.story_files_dir = ($output_dir + "/") |
+            .chrono_keeper.RecordingGroup = $recording_group |
+            .chrono_keeper.Logging.log.file = ($bin_dir + "/" + $j + "_" + .chrono_keeper.Logging.log.file)' "$default_conf" > "$output_file"
+        echo "Generated $output_file with ports $new_port_keeper_record and $new_port_keeper_datastore and $new_port_keeper_drain"
     done
 
-    local output_file="${conf_dir}/visor_conf.json"
+    # Generate visor configuration file
+    local visor_output_file="${conf_dir}/visor_conf.json"
     jq --arg bin_dir "$bin_dir" \
-       '.chrono_visor.Logging.log.file = ($bin_dir + "/" + .chrono_visor.Logging.log.file)' "$default_conf" > "$output_file"
+       '.chrono_visor.Logging.log.file = ($bin_dir + "/" + .chrono_visor.Logging.log.file)' "$default_conf" > "$visor_output_file"
+    echo "Generated $visor_output_file"
 
-    local output_file="${conf_dir}/client_conf.json"
+    # Generate client configuration file
+    local client_output_file="${conf_dir}/client_conf.json"
     jq --arg bin_dir "$bin_dir" \
-       '.chrono_client.Logging.log.file = ($bin_dir + "/" + .chrono_client.Logging.log.file)' "$default_conf" > "$output_file"
+       '.chrono_client.Logging.log.file = ($bin_dir + "/" + .chrono_client.Logging.log.file)' "$default_conf" > "$client_output_file"
+    echo "Generated $client_output_file"
 
+    local base_port_grapher_drain=$(jq '.chrono_grapher.KeeperGrapherDrainService.rpc.service_base_port' "$default_conf")
+    local base_port_grapher_datastore=$(jq '.chrono_grapher.DataStoreAdminService.rpc.service_base_port' "$default_conf")
+
+    # Generate grapher configuration files
+    echo "Generating grapher conf files ..."
+    mkdir -p "${output_dir}"
+    for (( i=1; i<num_recording_groups+1; i++ )); do
```

Comment:
pls use the default port numbers as the ports for the first instances of corresponding services, like 3333 as the first grapher drain service port.